blogs/lokiless/lokiless.md

```diff
@@ -0,0 +1,282 @@
+
+## A brief history
+
+When we started the NetObserv project, one of the first architectural questions was, as you could expect, which storage solution to adopt. It has to be robust for a write-intensive application, with indexing capabilities on large data sets, scalable, while still allowing to run complex queries. It must be able to store structured logs and to extract time-series from them. Features like full-text indexing or data mutability aren't required. On top of that, the license must be compatible with our needs. We ended up with a shortlist that included Grafana Loki, OpenDistro, Influx and a few others. This was two years ago.
+
+We also talked with other OpenShift teams having similar requirements, such as the Logging and the Distributed Tracing teams, and got some feedback to eventually rule out candidates with supposedly higher operational costs. And since the Logging team had already planned to invest in the Loki Operator, that was a nice opportunity to mutualize some efforts. Ok, let's be honest: that was a huge time saver especially for us, thanks so much folks!
+
+## Why change now?
+
+To be clear, **we aren't actually moving away from Loki**. Loki remains the one and only storage solution that we fully support at the moment, and our Console plugin is entirely based on queries to Loki, in its _logql_ format. However, we have seen some people using NetObserv in a way that we didn't expect: deploying it without Loki, and configuring flow exporters for instance with Kafka or IPFIX. Why? It turned out they were more interested in the kube-enriched raw flow data than in the visualizations that NetObserv provides, and they didn't want to deal with a new backend storage setup and maintenance. Which, admittedly, is a quite reasonable argument.
+
+To summarize, here's the deal:
+- 💰: you save on operational aspects by not deploying Loki or any storage that would be new to you.
+- 💔: you lose all the fancy dashboards that we build with so much love.
+- 👩🏽‍🔧: you need to create your own consumer — Kafka or IPFIX — for doing anything with the exported flows: this is on your plate. But you will see below that it's nothing insurmountable.
+
+## What is changed
+
+Before 1.4, there was still a problem: NetObserv was designed with Loki as a requirement. If you don't configure the Loki endpoint, our _flowlogs-pipeline_ component still tries to send flows to a default URL (and fails), and our Console plugin still tries to query Loki (and fails). While the latter isn't too annoying for someone who anyway doesn't want to use the Console plugin, the former could be the cause of performance degradation.
+
+So this is what we did: **we "just" added an _enable_ knob for Loki**. With Loki turned off, _flowlogs-pipeline_ obviously doesn't try to send anything to it. And since the Console plugin becomes useless without Loki, it isn't deployed anymore in that case.
```

Comment:
```suggestion
So this is what we did: **we "just" added an _enable_ knob for Loki**. With Loki turned off, the _flowlogs-pipeline_ obviously doesn't try to send anything to it. And since the console plugin becomes useless without Loki, it isn't deployed anymore in that case.
```

Reply:
Saying "the _flowlogs-pipeline_" sounds weird to my ears, but I don't know if it's just me, I know this is also used in the docs. Shouldn't we consider flowlogs-pipeline as a proper noun ?

Reply:
I think its OK to omit "the" if it sounds weird. I can see how it crowds the sentence. It could go either way I think..this was more of a suggestion. The is just a definite article used to increase clarity around a thing or object, and it could go before a proper noun but could also crowd the sentence if the meaning is general anyways.